src/clients/FtpClient.cc

```diff
@@ -914,6 +916,26 @@ Ftp::Client::maybeReadVirginBody()
 
     initReadBuf();
 
+    delayAwareRead();
+}
+
+void
+Ftp::Client::noteDelayedRead()
```

Comment:
We are notified about an opportunity to do a read, not about a (previously delayed) read that has happened now. I suggest clarifying this by renaming these methods into noteReadChance() or, if we must spell it out, noteDelayAwareReadChance().